x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/AbstractConvertFunction.java

```diff
@@ -53,6 +56,10 @@ protected AbstractConvertFunction(Source source, Expression field) {
         super(source, field);
     }
 
+    protected AbstractConvertFunction(StreamInput in) throws IOException {
+        this(Source.readFrom((PlanStreamInput) in), ((PlanStreamInput) in).readExpression());
```

Comment:
The second cast is temporary. The first one is the same thing @alex-spies has been talking about for a while - maybe there's a better way. But this way works for me, at least for now.